openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,116 @@
+import re
+import os
+import platform
+import PyOpenColorIO as ocio
+from openpype.settings import get_project_settings
+from openpype.pipeline import legacy_io
+from openpype.lib.log import Logger
+
+log = Logger.get_logger(__name__)
+
+
+def get_imagio_colorspace_from_filepath(
+    path, host=None, project_name=None, validate=True
```

Comment:
I would like to be able pass project settings. Also it would be better to get the settings only once when this function is called. Both `get_imageio_file_rules` and `get_imageio_config` are calling `get_project_settings`.